### PR TITLE
[DONT MERGE] WP 4ds loader

### PR DESCRIPTION
**Pull request by zpl-zak**

### CODE REVIEW

File: components/osg_loader.hpp [hunk 1]
```diff
@@ -12,43 +12,824 @@ class Loader
...
+		ProjectReflectTextureZ = 0x00004000,
+		AdditionalEffect = 0x00008000,
+		AlphaTexture = 0x40000000,
+		ColorKey = 0x20000000,
+		AdditiveMixing = 0x80000000
+	};
+
+};
+
+namespace MeshType
```
Comment (drummyfish) [inline components/osg_loader.hpp]: Better use typedefs instead of namespace for each enum?

Comment (Romop5) [reply components/osg_loader.hpp]: Isn't this *namespace version* equal to *enum class* ? See http://en.cppreference.com/w/cpp/language/enum

Comment (drummyfish) [reply components/osg_loader.hpp]: Might be, anyway either use the enum class or typedef, that way we won't have a billion namespaces in doxygen docs etc. - I'd go for C++14 features only if there's a clear benefit though, i.e. typedefs might achieve the same while keeping the module work with lower C++ versions.

**Comment by drummyfish** [comment]
I suppose this is only a first draft, but just to be clear: the code you added should be moved to format_parsers.hpp to a subclass of the format class. That class should then be used in osg_loaders.hpp to get the model data and construct the OSG node.

**Comment by inlife** [comment]
are you sure that returning a string is a good idea? doesn't seem scalable
seems to me, that there might be a lot of occasions where you would call log from within the command
and just return an empty string to the output, or a lot of concatenation

**Pull request merged**